### ISSUE TITLE
Delayed output to us not being claimed.

### ISSUE THREAD

**Issue by alecalve** [ElementsProject/lightning#747]
A while back, one of my channels was unilaterally closed by lightningd while I was fiddling with the RPC commands (I'm still not sure what happened there).

The channel went in the `ONCHAIND_OUR_UNILATERAL` state, which from my understanding meant I had to wait `to_self_delay` blocks (144) before my node would be able to claim its output, which would then be visible in `listfunds`. However, the transaction has now 365 confirmations and the delayed output is still unspent.

Is something amiss with my node or is my understanding of the situation incorrect?

**Comment by ZmnSCPxj** [comment]
Something does seem amiss.  I think `onchaind` should have been monitoring the situation and should have noticed the closing transaction was deep enough.

**Comment by alecalve** [comment]
To be sure, I tried closing another channel, and this time everything went smoothly :thinking:

**Comment by cdecker** [comment]
`onchaind` should try to continue were it left off when restarting, did you try that?

**Comment by alecalve** [comment]
I did restart the node, checking `dev-blockheight` while it was starting.

It started before the channel closure, then blew past it without touching the output. Here are some relevant log lines:

```
lightningd(7830): peer 02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432: Peer permanent failure in ONCHAIND_OUR_UNILATERAL: Funding transaction spent
lightningd(7830):        (tx 589e0846aff86da6a0e7179ce00957a8b95f081ca7d0729b93241763927c6c21)
lightningd(7830): peer 02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432: state: ONCHAIND_OUR_UNILATERAL -> FUNDING_SPEND_SEEN
lightningd(7830): peer 02f6725f9c1c40333b67faea92fd211c183050f28df32cac3f9d69685fe9665432: state: FUNDING_SPEND_SEEN -> ONCHAIND_OUR_UNILATERAL
```

**Comment by cdecker** [comment]
Do you see entries like the following in your logs?

```
Propose handling OUR_UNILATERAL/DELAYED_OUTPUT_TO_US by OUR_DELAYED_RETURN_TO_WALLET
```

**Comment by cdecker** [comment]
I appears that the delayed output was indeed be spent here: b98bb881178a786881d0da3d5466ee59588a048ac2f5453c69adfa5a3f194373 with 3 confirmations. So I appears that `lightningd` picked up the delayed output?

**Comment by alecalve** [comment]
Oh, I didn't notice it. It seems that several restarts did the trick and that I was too hasty in judging `lightningd` ignored it. I didn't notice that log line, but I'm not sure it didn't happen as the logs went past my terminal's history. Once my node is fully synced, I'll check if the funds appear in `listfunds`.

**Comment by alecalve** [comment]
It might be that the fix to #709 was the solution as I recently pulled master and restarted my node.

**Comment by alecalve** [comment]
The funds are now listed in `listfunds`
